### PR TITLE
Add format option to net block

### PR DESCRIPTION
**Pull request by gurditsbedi**
PR for the issue #692 . 
The format option is added to the `net` block by this PR.
If a user uses a deprecated option, the `format` option will not be used. This helps users into transitioning to the new options.

### CODE REVIEW

**Comment by ammgws** [comment]
Might need to check the logic.

Using this config on my wired desktop:
```toml
[[block]]
block = "net"
```

Get this: `Error in block 'net': Signal strength is only available for connected wireless devices.`

**Comment by gurditsbedi** [comment]
Confirming the error. Will look into it.

**Comment by ammgws** [comment]
Did you figure it out?

**Comment by gurditsbedi** [comment]
> Did you figure it out?

I think 83a8311f956ad2efb09bf2d1888b20a0988d7bed fixed it. Is the issue still not resolved?

**Comment by gurditsbedi** [comment]
Looks like, I did a silent commit and I didn't announce the update in the thread.

**Comment by ammgws** [comment]
Can you fix the blocks.md conflict?

**Comment by gurditsbedi** [comment]
I think its done. 
Sorry I am new to this, so I thought that the "merger to master" resolve the conflicts, that to for no reason at all.

**Comment by gurditsbedi** [comment]
I see that the last issue was relating to line endings. "\n" and "\r\n". 
I think we should decide upon the type of line ending to be used in the project.

Also we can do the following
1. checking of line endings in the CI pipeline.
OR
2. add a git hook commit script.

**Comment by ammgws** [comment]
They should all be Unix line endings. I fixed the conflict for now.

File: src/util.rs [hunk 1]
```diff
@@ -334,6 +335,36 @@ pub fn format_percent_bar(percent: f32) -> String {
         .collect()
 }
 
+pub fn format_vec_to_bar_graph<T>(content: &[T], min: Option<T>, max: Option<T>) -> String
+where
+    T: Ord + ToPrimitive,
+{
+    let bars = [
+        '\u{2581}', '\u{2582}', '\u{2583}', '\u{2584}', '\u{2585}', '\u{2586}', '\u{2587}',
```
Comment (ammgws) [inline src/util.rs:343]: Can you add a comment saying what these unicode chars are meant to be?

File: src/blocks/net.rs [hunk 2]
```diff
@@ -854,41 +867,52 @@ impl Block for Net {
...
+            "{speed_down}" => &s_dn,
+            "{graph_up}" =>  self.graph_tx.as_ref().unwrap_or(&empty_string),
+            "{graph_down}" =>  self.graph_rx.as_ref().unwrap_or(&empty_string)
+        );
+
+        self.output
+            .set_text(if let Ok(s) = self.format.render_static_str(&values) {
+                s
+            } else {
+                "[invalid connection format string]".to_string()
```
Comment (ammgws) [inline src/blocks/net.rs:907]: I believe we usually return a BlockError here. Is there a reason for deviating from that pattern?

Comment (gurditsbedi) [reply src/blocks/net.rs]: I was refering to https://github.com/greshake/i3status-rust/blob/master/src/blocks/networkmanager.rs file while doing. After you mentioned this, I realized that similar problem exists in that file also.  Also I have update the file as per this comment.

File: src/util.rs (hunk 1 above)
Comment (gurditsbedi) [reply src/util.rs:344]: ` (x * one eighth block) https://en.wikipedia.org/wiki/Block_Elements`
Added to the file also.

**Comment by ammgws** [comment]
Thanks!

**Pull request merged**